TESTS/mbed_hal_fpga_ci_test_shield/spi/main.cpp

```diff
@@ -177,9 +177,9 @@ Case cases[] = {
     Case("SPI - mode testing (MODE_1)", one_peripheral<SPIPort, DefaultFormFactor, spi_test_common<SPITester::Mode1, 8, TRANSFER_SPI_MASTER_WRITE_SYNC, FREQ_1_MHZ> >),
     Case("SPI - mode testing (MODE_2)", one_peripheral<SPIPort, DefaultFormFactor, spi_test_common<SPITester::Mode2, 8, TRANSFER_SPI_MASTER_WRITE_SYNC, FREQ_1_MHZ> >),
     Case("SPI - mode testing (MODE_3)", one_peripheral<SPIPort, DefaultFormFactor, spi_test_common<SPITester::Mode3, 8, TRANSFER_SPI_MASTER_WRITE_SYNC, FREQ_1_MHZ> >),
-
+#if !defined(TARGET_NRF52840_DK)
```

Comment:
Why is this test skipped just for the NRF52840? Is this test tied to a defined behavior in the SPI HAL specification? If so users and library may be relying on all targets having this behavior. If this isn't part of the HAL specification then maybe this could be an optional test? What do you think?

Reply:
Thanks for the review!
16-bit symbols are not supported on `NRF` target. On master, specification is not very restrictive since it must be consistent with all targets. We also don't have capabilities like on the SPI feature branch, so I can't run this case conditionally. 

I see two options:
- remove the test case permanently, but it would be good to run this test on targets that support 16-bit symbols.
- use the preprocessor to disable the test case on the specific target.

I don't like the first option. Let's consider that we are adding the next target which supports only one clock mode. If we choose this approach, then I would have to remove test cases for other modes. 

Also on the feature branch, we have nice specification and tests, but it looks like there are no plans to bring new SPI into the master in the nearest future 😞 Maybe it would be good to add `spi_get_capabilities()` function on the master.

Reply:
> Also on the feature branch, we have nice specification and tests, but it looks like there are no plans to bring new SPI into the master in the nearest future 😞 Maybe it would be good to add spi_get_capabilities() function on the master.

How much work it would be?

Reply:
We could add `WEAK` definition for all targets which returns that all SPI capabilities are supported. And then implement a valid version for CI targets and disable unsupported capabilities for each CI target. This way we could also extend the SPI test on the master. I think that this should be easy to implement.

Reply:
I've realized just now this has not yet been fixed?

Reply:
Not at the moment. This must be discussed. I'll try to write an email about this. The same problem we have already here:
https://github.com/ARMmbed/mbed-os/blob/df79609cc56c063ba292972af77f2b833b0fda35/TESTS/mbed_hal_fpga_ci_test_shield/gpio/main.cpp#L67-L72